packages/utils/formatUnit/formatUnit.ts

```diff
@@ -0,0 +1,126 @@
+type Unit = "K" | "M" | "B" | "T" | "";
+type Sign = "-" | "";
+
+type Config = {
+  decimalPlaceToRoundAt?: number;
+  maxDecimalPlaces?: number;
+  forcedUnit?: Unit;
+};
+
+/**
+ * This is a function that shortens a number into appropriate units (supports until trillion).
+ *
+ * Ex: formatUnit(100_000_000) => "1M"
+ *
+ * Ex: formatUnit(155_555_555, { decimalPlaceToRoundAt: 2, maxDecimalPlaces: 3 }) => "155.560M"
+ *
+ * @param {number} value The number to shorten.
+ * @param {Object} [config] Optional
+ * @param {number | undefined} [config.decimalPlaceToRoundAt] The decimal place to round at. Optional.
+ * @param {number | undefined} [config.maxDecimalPlaces] The maximum number of decimal places to show. Optional.
+ * @param {Unit | undefined} [config.forcedUnit] The unit to force. Optional.
+ */
+
+export default function formatUnit(value: number, config?: Config) {
+  const { decimalPlaceToRoundAt, maxDecimalPlaces, forcedUnit } = config || {};
+
+  const absValue = Math.abs(value);
+
+  if (absValue > Number.MAX_SAFE_INTEGER) {
+    throw Error("The given value is too large or too small to safely handle.");
+  }
+  if (
+    (decimalPlaceToRoundAt && decimalPlaceToRoundAt < 0) ||
+    (maxDecimalPlaces && maxDecimalPlaces < 0)
+  ) {
+    throw Error("Decimal places must be a positive number.");
+  }
+  if (
+    decimalPlaceToRoundAt &&
+    maxDecimalPlaces &&
+    decimalPlaceToRoundAt > maxDecimalPlaces
+  ) {
+    throw Error(
+      "The decimal place to round at cannot be greater than the number of decimal places."
+    );
+  }
+
+  const sign: Sign = value < 0 ? "-" : "";
+
+  if (forcedUnit) {
+    return abbreviate(
+      absValue,
+      decimalPlaceToRoundAt,
+      maxDecimalPlaces,
+      forcedUnit,
+      sign
+    );
+  }
```

Comment:
Instead of this part, how about changing it as follows in the return?
```tsx
  return abbreviate(
    absValue,
    decimalPlaceToRoundAt,
    maxDecimalPlaces,
    forcedUnit ? forcedUnit : unit,
    sign
  );
```